### PR TITLE
Improved logging for Api Client.get NewAccessToken

### LINKED ISSUE #79: ApiClient Errors are catched without throwing or logging the real Exceptions
## Summary

When using `ApiClient.getNewAccessToken()`, the real cause of the error is not returned.(Instead the is System.out.println)
https://github.com/cyberark/conjur-sdk-java/blob/8b6739b9b8d25966569ef7956c78e17659b10269/client/src/main/java/com/cyberark/conjur/sdk/ApiClient.java#L1284

This causes, many issues while trying to troubleshoot root cause.
https://github.com/cyberark/conjur-sdk-java/blob/8b6739b9b8d25966569ef7956c78e17659b10269/client/src/main/java/com/cyberark/conjur/sdk/ApiClient.java#L1304-L1307

It's impossible with the current implementation, to get get any relevant information about the root cause of the error from the client logs.

## Steps to Reproduce

1. Call the `ApiClient.getNewAccessToken()` when the conjur instance with **wrong credentials** (host id / API KEY)
2. See error in the logs: **Failed to automatically update AccessToken**

## Expected Results

The root cause of the error should be logged, so the client knows: 
- The http code returned from the conjur server
- The error message

## Actual Results

Message in the logs: **Failed to automatically update AccessToken**
Screenshot 2023-04-11 at 20 19 25

## Reproducible

   * [x] Always 
   * [ ] Sometimes
   * [ ] Non-Reproducible
   
## Version/Tag number

**com.cyberark:conjur-sdk-java:4.0.0**

## Environment setup

* Running on macOS 13.2.1 / IntelliJ IDEA 2022.2
* Java version: 11.0.15
* conjur enterprise 12.9

## Additional Information

* This issue impacts the [conjur-spring-boot-sdk]( https://github.com/cyberark/conjur-spring-boot-sdk).

### PR DESCRIPTION
**Pull request by bnasslahsen**
Closes #79 

### Desired Outcome

* The objective of this PR is to log real cause of errors, when using `ApiClient.getNewAccessToken() `

### Implemented Changes

 Logging using the JDK Logger:
- Status code
- The exception message
- The cause of the error

### Connected Issue/Story

Resolves #79

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: client/src/main/java/com/cyberark/conjur/sdk/ApiClient.java [hunk 1]
```diff
@@ -1302,7 +1304,9 @@ public AccessToken getNewAccessToken() {
             ApiResponse<String> response = execute(c, new TypeToken<String>(){}.getType());
             accessToken = response.getData();
         } catch (ApiException e) {
-            System.out.println("Failed to automatically update AccessToken");
+            LOGGER.severe("Status code: " + e.getCode());
+            LOGGER.severe("Reason: " + e.getResponseBody());
+            LOGGER.severe(e.getMessage());
```
Comment (maciek01) [inline client/src/main/java/com/cyberark/conjur/sdk/ApiClient.java:1309]: why are we not logging actual stack trace?

**Comment by john-odonnell** [comment]
This Java SDK is generated from [Conjur's OpenAPI description](https://github.com/cyberark/conjur-openapi-spec). When making functional changes, instead of updating the `.java` files directly, we need to:
1. Update our set of Mustache templates (specifically [here](https://github.com/cyberark/conjur-openapi-spec))
2. Refresh the `.java` files using [`./bin/refresh_client`](https://github.com/cyberark/conjur-sdk-java/blob/main/bin/refresh_client)

If we update the `.java` files without updating the Mustache templates, the changes will be reverted on the next release.

File: client/src/main/java/com/cyberark/conjur/sdk/ApiClient.java [hunk 2]
```diff
@@ -1302,7 +1304,9 @@ public AccessToken getNewAccessToken() {
             ApiResponse<String> response = execute(c, new TypeToken<String>(){}.getType());
             accessToken = response.getData();
         } catch (ApiException e) {
-            System.out.println("Failed to automatically update AccessToken");
```
[the remark this answers is not in the document]
Comment (bnasslahsen) [reply client/src/main/java/com/cyberark/conjur/sdk/ApiClient.java:1305]: i will update the error message to add the context of the error.

File: client/src/main/java/com/cyberark/conjur/sdk/ApiClient.java (hunk 1 above)
Comment (bnasslahsen) [reply client/src/main/java/com/cyberark/conjur/sdk/ApiClient.java:1309]: Logging the stack trace seemed to be like a bonus. 
These informations are the minimal information required to troubleshoot.
But definitely agree that nothing better than logging the stackTrace.

**Comment by bnasslahsen** [comment]
@john-odonnell,

The corresponding Mustache template is now updated. 
I have pushed the version of the generated java ApiClient as well.

**Comment by john-odonnell** [comment]
Looks like the integration tests are failing - I'll look into these before approving, but I don't anticipate they're caused by anything you've done here.

**Comment by bnasslahsen** [comment]
@john-odonnell,

The rebase is done. 
I can see that the the integration tests are passing now.

For the `codeclimate` issue, it's related to `bin/test_integration` which is not part of the PR files.

**Review by jtuttle — APPROVED** [review]

**Pull request merged**